pkg/ns/ns_linux.go

```diff
@@ -152,6 +152,41 @@ func GetNS(nspath string) (NetNS, error) {
 	return &netNS{file: fd}, nil
 }
 
+// Returns a new empty NetNS.
+func TempNetNS() (NetNS, error) {
+	// save a handle to current network namespace
+	hostNS, err := GetCurrentNS()
+	if err != nil {
+		return nil, fmt.Errorf("failed to open current namespace: %v", err)
+	}
+	defer hostNS.Close()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	var ns NetNS
+
+	go func() {
```

Comment:
why spawn a goroutine if you are blocking until it finish and no other code runs?

Reply:
Reworked a bit the function and added comments, it's the same as Do(), if we fail to switch the namespace back we leave the thread to die